components/styled/table/src/molecules/table-methods.tsx

```diff
@@ -100,4 +103,114 @@ const prependColumnWithExpansionControls = (
   return [expanderColumn].concat(columns)
 }
 
-export default createColumns
+const calculateStaticColumnOffset = (cellIdx: number, staticColumns: number, firstColumnWidth: number) => {
+  if (staticColumns === 0) {
+    return undefined
+  }
+
+  if (cellIdx === 0) {
+    return 0
+  }
+
+  if (cellIdx < staticColumns) {
+    return firstColumnWidth
+  }
+
+  return undefined
+}
+
+const handleScrollEvent = (table: Table<DataEntity>, headerAxis: Axis, scrollState: ScrollState) => {
+  const { scrollWidth, scrollHeight, currentXScroll, currentYScroll } = scrollState
+  if (
+    headerAxis === 'x' &&
+    currentYScroll > scrollHeight - 20 &&
+    currentYScroll < scrollHeight + 20 &&
+    table.getCanNextPage()
+  ) {
+    table.nextPage()
+  }
+
+  if (
+    headerAxis === 'y' &&
+    currentXScroll > scrollWidth - 20 &&
+    currentXScroll < scrollWidth + 20 &&
+    table.getCanNextPage()
+  ) {
+    table.nextPage()
+  }
+}
+
+const handleScrollEventManual = (
+  getCanNextPage: () => boolean,
+  nextPage: () => void,
+  headerAxis: Axis,
+  scrollState: ScrollState
+) => {
+  const { scrollWidth, scrollHeight, currentXScroll, currentYScroll } = scrollState
+  if (
+    headerAxis === 'x' &&
+    currentYScroll > scrollHeight - 20 &&
```

Comment:
Is the number 20 being computed from anything here or just a guestimate? If it's just a guess it might be better to derive it as a % of the max unless it's too much work - could possibly abstract these 4 repeated blocks then too